docs/NNNN-schema-registry.md

```diff
@@ -0,0 +1,195 @@
+
+* Name: extensionName
+  * Description: String identifying the extension.
+  * Type: String
+  * Constraints: Must be "NNNN-schema-registry".
+  * Default: "NNNN-schema-registry"
+
+* Name: extensionVersion
+  * Description: String identifying extension version.
+  * Type: String
+  * Constraints: Must be a valid published version of this specification.
+  * Default: "0.1"
+
+* Name: identifierDigestAlgorithm
+  * Description: Algorithm used for calculating safe filenames from schema identifiers.
+  * Type: String
+  * Constraints: Must be a valid digest algorithm returning strings which are safe file names on the target file system.
+  * Default: "md5"
+
+* Name: digestAlgorithm
+  * Description: Digest algorithm used for calculating fixity of the schema inventory and stored schema files.
+  * Type: String
+  * Constraints: Must be a valid digest algorithm.
+  * Default: The same value use elsewhere in the OCFL for integrity checking.
+
+An example config.json in included in the Examples section below.
+
+## schema_inventory.json
+
+A manifest of the registered schemata must be maintained in schema_inventory.json. Implementations of this extension must treat integrity verification of the stored schemata with the same priority as all primary OCFL objects. 
+
+* Name: manifest
+  * Description: Object containing manifest entries. Indexed by the stored filename - ie the digest of the schema's identifier.
+  * Type: Object
+  * Constraints: Must contain one entry per registered schema; must contain 2 sub-keys documented below.
+  * Default: Not applicable
+
+### manifest entry properties for schema_inventory.json
+
+* Name: digest
+  * Description: The digest of the stored schema file, used for integrity verification.
+  * Type: String
+  * Constraints: Must be the digest of the stored file, as calculated using the configured digestAlgorithm.
+  * Default: Not applicable
+
+* Name: identifier
+  * Description: Original identifier string for the schema.
+  * Type: String
+  * Constraints: Must be the string that was hashed with identifierDigestAlgorithm to produce the stored file name.
+  * Default: Not applicable
+
+## Implementation
+
+In an OCFL root where this extension is configured, tools reading the OCFL objects may be aware of local copies of schemata. Accessing local schemata may reduce latency and provide a performance benefit. Thus, OCFL reading applications relying on schemata (eg for validation) may wish to read these from the schema registry.
+
+A process must be implemented which ensures all OCFL objects/versions written are checked for schema references. Where not already registered, the schema must be retrieved and registered as described below. Since schemata are typically identified by the URL of the specific version referenced, the digest of the schema's identifier (as configured by identifierDigestAlgorithm) must be used as a filename. MD5 may be sufficient although other algorithms may be configured. 
```

Comment:
I see two parts to this proposal:

1) How to store schemata in an OCFL root
2) Validating and ensuring that schemata references in all OCFL objects is archived.

I would recommend only being prescriptive about how to store the schemata in an OCFL root. Regarding how to extra those schema references from OCFL objects, I'd rather simply state something like, "when writing OCFL objects you SHOULD ensure that all schemata references are extracted and stored". I feel how you extract references, which ones you want to extract, and which ones are relevant important, should be left up to the one writing an OCFL object into a root.